### PR TITLE
fixed-39

### LINKED ISSUE #39: Make non-successful responses accept a string
It doesn't make sense to accept a `map[string]interface{}` for a response which doesn't return data, only a status message.

So far the only function affected is `SendUnauthorized`.

Acceptance criteria:
- `SendUnauthorized` should accept `err string` instead of `data map[string]interface{}`
- implement changes and create PR(s) in affected applications, the first fix should be in https://github.com/RedHatInsights/insights-results-aggregator/blob/master/server/auth.go at about  ~L90 where a token is missing, thus should be `Unauthorized` instead of `Forbidden`. There should be a `TODO:` comment soon

### PR DESCRIPTION
**Pull request by Sergey1011010**
Fixes #39 

## Type of change

Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)
- Refactor (refactoring code, removing useless files)
- Unit tests

## Testing steps

`make style; and make test`

### CODE REVIEW

**Review by tisnik — APPROVED** [review]
OK

File: types/errors.go [hunk 1]
```diff
@@ -80,7 +80,7 @@ func HandleServerError(writer http.ResponseWriter, err error) {
 	case *ItemNotFoundError:
 		respErr = responses.SendNotFound(writer, err.Error())
 	case *AuthenticationError:
-		respErr = responses.SendForbidden(writer, err.Error())
+		respErr = responses.SendUnauthorized(writer, err.Error())
```
Comment (Bee-lee) [inline types/errors.go:83]: I dont think `AuthenticationError` always means Unauthorized. 
Forbidden is a valid response for some places where `AuthenticationError` is used

**Comment by Bee-lee** [comment]
`SendUnauthorized` is also used here https://github.com/RedHatInsights/insights-operator-ldapauth/blob/master/server/auth.go#L45

File: types/errors.go (hunk 1 above)
Comment (Sergey1011010) [reply types/errors.go:83]: As far as I can see, we use it only when reporting problems with authentication, we can add another type of error tho

Comment (Bee-lee) [reply types/errors.go:83]: OK. Also are we sure that OCM or anything else isn't relying on the response codes? Just checking

**Review by Bee-lee — APPROVED** [review]

File: types/errors.go (hunk 1 above)
Comment (Sergey1011010) [reply types/errors.go:93]: What do you think about the new changes?

Comment (Bee-lee) [reply types/errors.go:93]: makes more sense to me, thx!

**Pull request merged**